cloudevents/encoding_binary.go

```diff
@@ -147,13 +147,11 @@ func (binary) ToHeaders(context *EventContext, h http.Header) error {
 		return err
 	}
 	// Defaultable values:
-	ceVersion := context.CloudEventsVersion
-	if ceVersion == "" {
-		ceVersion = CloudEventsVersion
+	if context.CloudEventsVersion == "" {
+		context.CloudEventsVersion = CloudEventsVersion
 	}
-	contentType := context.ContentType
-	if contentType == "" {
-		contentType = contentTypeJSON
+	if context.ContentType == "" {
+		context.ContentType = contentTypeJSON
```

Comment:
You should document that you are changing the context supplied as an argument, which is surprising. Otherwise, you want to make a copy of the context before writing to it here.

(You should probably write a test to verify that the the supplied arguments to a binary transform are not themselves modified by the call.)

Reply:
they are being modified, and this lack of defaulting is a bug in the current code, not setting the content type breaks binary encoding because golang can't encode empty content type